collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py

```diff
@@ -754,6 +826,57 @@ def generate_pgroups_dict(array):
     return pgroups_info
 
 
+def generate_rl_dict(array):
+    rl_info = {}
+    api_version = array._list_available_rest_versions()
+    if ACTIVE_DR_API in api_version:
+        rlinks = array.list_pod_replica_links()
+        for rlink in range(0, len(rlinks)):
+            link_name = rlinks[rlink]['local_pod_name']
+            since_epoch = rlinks[rlink]['recovery_point'] / 1000
+            recovery_datatime = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(since_epoch))
+            rl_info[link_name] = {
+                'status': rlinks[rlink]['status'],
+                'direction': rlinks[rlink]['direction'],
+                'lag': str(rlinks[rlink]['lag'] / 1000) + 's',
+                'remote_pod_name': rlinks[rlink]['remote_pod_name'],
+                'remote_names': rlinks[rlink]['remote_names'],
+                'recovery_point': recovery_datatime,
+            }
+    return rl_info
+
+
+def generate_del_pods_dict(array):
+    pods_info = {}
+    api_version = array._list_available_rest_versions()
+    if AC_REQUIRED_API_VERSION in api_version:
+        pods = array.list_pods(mediator=True, pending_only=True)
+        for pod in range(0, len(pods)):
+            acpod = pods[pod]['name']
+            pods_info[acpod] = {
+                'source': pods[pod]['source'],
+                'arrays': pods[pod]['arrays'],
+                'mediator': pods[pod]['mediator'],
+                'mediator_version': pods[pod]['mediator_version'],
+            }
+            if pods_info[acpod]['arrays'][0]['frozen_at']:
```

Comment:
Suggest to change `pods_info[acpod]['arrays'][0]['frozen_at']` to `pods_info[acpod]['arrays'][0].get('frozen_at')` for backwards compatibility with purity 5.x.

Reply:
Moved to a better location where version check is applied